File: backend/CncApp/CncApp.Application/Dtos/Jobs/JobProductionDto.cs
using CncApp.Application.Dtos.Shifts;

namespace CncApp.Application.Dtos.Jobs;

public class JobProductionDto
{
    public int Id { get; set; }

    public int OrderId { get; set; }

    public DateOnly DueDate { get; set; }

    public int MachineId { get; set; }

    public string MachineName { get; set; } = string.Empty;

    public int PartAmountPlanned { get; set; }

    public string PartName { get; set; } = string.Empty;

    public string PartNumber { get; set; } = string.Empty;

    public int PartsCompleted { get; set; }

    public decimal PercentComplete { get; set; }

    public int? StockLotId { get; set; }

    public string? LotNumber { get; set; }

    public List<ShiftDto> Shifts { get; set; } = new();
}

File: frontend/angular/src/app/core/dtos/jobs/job-production.dto.ts
import { ShiftDto } from '../shifts/shift.dto';

export interface JobProductionDto {
  id: number;
  orderId: number;
  dueDate: string;
  machineId: number;
  machineName: string;
  partAmountPlanned: number;
  partName: string;
  partNumber: string;
  partsCompleted: number;
  percentComplete: number;
  shifts: ShiftDto[];
  stockLotId: number | null;
  lotNumber: string | null;
}
